<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="prata" />


<title>MajorityVoteDevelopment.R</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">MajorityVoteDevelopment.R</h1>
<h4 class="author"><em>prata</em></h4>
<h4 class="date"><em>Thu Jan 17 16:48:53 2019</em></h4>



<p>Here we describe the development of our functions to McIntyre et al. (2017) Majority Vote Ensemble Classifier.</p>
<div id="preliminares" class="section level1">
<h1>Preliminares</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># setwd(&quot;Rmd&quot;)</span>
<span class="kw">getwd</span>()</code></pre></div>
<pre><code>## [1] &quot;/home/prata/projects/gunga/Rmd&quot;</code></pre>
<p>Knitr options:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span>opts_current<span class="op">$</span><span class="kw">set</span>(<span class="dt">cache =</span> <span class="ot">TRUE</span>)
knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">cache =</span> <span class="ot">TRUE</span>)
knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Libraries:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(dplyr)</code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># library(ggplot2)</span>
<span class="kw">library</span>(magrittr)
<span class="kw">library</span>(methods)
<span class="kw">library</span>(stringr)</code></pre></div>
<p>Taxdumpr object to taxonomy processing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(taxdumpr)
taxdumprObject &lt;-<span class="st"> </span><span class="kw">Taxdumpr</span>(<span class="st">&quot;~/taxdump/nodes.dmp&quot;</span>, <span class="st">&quot;~/taxdump/names.dmp&quot;</span>, <span class="st">&quot;~/taxdump/merged.dmp&quot;</span>)</code></pre></div>
</div>
<div id="profiles-loading" class="section level1">
<h1>Profiles Loading</h1>
<div id="diamond-megan" class="section level2">
<h2>Diamond-MEGAN</h2>
<p>Functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loadMeganClassifications &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">meganClassificationsLocation =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  ##
  classificationsDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_lines</span>(meganClassificationsLocation) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">lineage =</span> .)
  classificationsDf<span class="op">$</span>sequenceId &lt;-<span class="st"> </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_extract</span>(<span class="st">&quot;.*?;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;;&quot;</span>, <span class="st">&quot;&quot;</span>)


  ##
  rawTaxonomy &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;[dpcofgs]__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  classificationsDf<span class="op">$</span>name &lt;-<span class="st"> </span>rawTaxonomy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(first) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;[dpcofgs]__&quot;</span>, <span class="st">&quot;&quot;</span>)
  classificationsDf<span class="op">$</span>score &lt;-<span class="st">  </span>rawTaxonomy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric

  taxonomyDf &lt;-<span class="st"> </span><span class="ot">NULL</span>
  meganTaxonomyLocation &lt;-<span class="st"> </span>meganClassificationsLocation <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;(.tsv.gz$|.tsv$)&quot;</span>, <span class="st">&quot;.taxonomy.tsv.gz&quot;</span>)
  classificationsDf<span class="op">$</span>taxonomyId &lt;-<span class="st">  </span>classificationsDf<span class="op">$</span>name <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardTaxonomyIdsByNames</span>(taxdumprObject, .)
  <span class="co"># if (file.exists(meganTaxonomyLocation)) {</span>
    <span class="co"># taxonomyDf &lt;- readr::read_tsv(meganTaxonomyLocation)</span>
  <span class="co"># } else {</span>
    taxonomyDf &lt;-<span class="st">  </span>classificationsDf<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardLineageIdsAndScientificNamesByIdsAsDataFrame</span>(taxdumprObject, .)
    <span class="co"># write.table(taxonomyDf, gzfile(meganTaxonomyLocation), quote = F, sep = &quot;\t&quot;, row.names = F, col.names = T)</span>
  <span class="co"># }</span>
  classificationsDf &lt;-<span class="st"> </span>classificationsDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">merge</span>(taxonomyDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all.x =</span> T)

  ##
  rawSuperkingdom &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;d__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$superkingdom &lt;- rawSuperkingdom %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>superkingdomScore &lt;-<span class="st"> </span>rawSuperkingdom <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric

  ##
  rawPhylum &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;p__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$phylum &lt;- rawPhylum %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>phylumScore &lt;-<span class="st"> </span>rawPhylum <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric


  ##
  rawClass &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;c__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$class &lt;- rawClass %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>classScore &lt;-<span class="st"> </span>rawClass <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric


  ##
  rawOrder &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;o__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$order &lt;- rawOrder %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>orderScore &lt;-<span class="st"> </span>rawOrder <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric

  ##
  rawFamily &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;f__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$family &lt;- rawFamily %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>familyScore &lt;-<span class="st"> </span>rawFamily <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric


  ##
  rawGenus &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;g__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$genus &lt;- rawGenus %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>genusScore &lt;-<span class="st"> </span>rawGenus <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric


  ##
  rawSpecies &lt;-
<span class="st">    </span>classificationsDf<span class="op">$</span>lineage <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_extract_all</span>(<span class="st">&quot;s__.*?;.*?;&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">sapply</span>(., last) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">str_replace_all</span>(<span class="st">&quot;;$&quot;</span>, <span class="st">&quot;&quot;</span>)
  <span class="co"># classificationsDf$species &lt;- rawSpecies %&gt;% str_split(&quot;;&quot;) %&gt;% sapply(first) %&gt;% str_replace(&quot;[dpcofgs]__&quot;, &quot;&quot;)</span>
  classificationsDf<span class="op">$</span>speciesScore &lt;-<span class="st"> </span>rawSpecies <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_split</span>(<span class="st">&quot;;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sapply</span>(last) <span class="op">%&gt;%</span><span class="st"> </span>trimws <span class="op">%&gt;%</span><span class="st"> </span>as.numeric


  classificationsDf &lt;-<span class="st"> </span>classificationsDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>lineage)
  <span class="kw">return</span>(classificationsDf)
}

getMeganProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">classifications =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  numberOfSequences &lt;-<span class="st"> </span><span class="kw">nrow</span>(classifications)
  profile1 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> superkingdomId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(superkingdomScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(superkingdomScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(superkingdomScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(superkingdomScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(superkingdomScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;superkingdom&quot;</span>)

  profile2 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> phylumId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(phylumScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(phylumScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(phylumScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(phylumScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(phylumScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;phylum&quot;</span>)

  profile3 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> classId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(classScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(classScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(classScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(classScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(classScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;class&quot;</span>)

  profile4 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> orderId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(orderScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(orderScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(orderScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(orderScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(orderScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;order&quot;</span>)

  profile5 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> familyId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(familyScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(familyScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(familyScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(familyScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(familyScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;family&quot;</span>)

  profile6 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> genusId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(genusScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(genusScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(genusScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(genusScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(genusScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;genus&quot;</span>)

  profile7 &lt;-
<span class="st">    </span>classifications <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">standardId =</span> speciesId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(standardId) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarize</span>(<span class="dt">MeganPercentual =</span> <span class="kw">n</span>() <span class="op">/</span><span class="st"> </span>numberOfSequences,
              <span class="dt">MeganSequencesNumber =</span> <span class="kw">n</span>(),
              <span class="dt">MeganScoreMean =</span> <span class="kw">mean</span>(speciesScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreStandardDeviation =</span> <span class="kw">sd</span>(speciesScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreIQR =</span> <span class="kw">IQR</span>(speciesScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreMedian =</span> <span class="kw">median</span>(speciesScore, <span class="dt">na.rm =</span> T),
              <span class="dt">MeganScoreCI =</span> <span class="kw">qnorm</span>(<span class="fl">0.975</span>) <span class="op">*</span><span class="st"> </span><span class="kw">sd</span>(speciesScore, <span class="dt">na.rm =</span> T) <span class="op">/</span><span class="st"> </span><span class="kw">sqrt</span>(<span class="kw">n</span>())) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">rank =</span> <span class="st">&quot;species&quot;</span>)

  profile &lt;-<span class="st"> </span><span class="kw">rbind</span>(profile1,profile2,profile3,profile4,profile5,profile6,profile7)
  profile<span class="op">$</span>MeganPercentual &lt;-<span class="st"> </span>profile<span class="op">$</span>MeganPercentual <span class="op">*</span><span class="st"> </span><span class="dv">100</span>
  <span class="kw">return</span>(profile)
}

<span class="co"># list.files(&quot;../data-raw/sh&quot;, &quot;megan_classifications.lca.tsv.gz&quot;)</span>
<span class="co"># meganClassificationsLocation &lt;- list.files(&quot;../data-raw/sh&quot;, &quot;megan_classifications.lca.tsv.gz&quot;, full.names = T)</span>
<span class="co"># meganClassificationsDf &lt;- loadMeganClassifications(meganClassificationsLocation, taxdumprObject)</span>
<span class="co"># head(meganClassificationsDf %&gt;% subset(speciesId %&gt;% is.na == F))</span>
<span class="co"># nrow(meganClassificationsDf)</span>
<span class="co"># meganProfileDf &lt;- getMeganProfile(meganClassificationsDf)</span>
meganProfileDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../data-raw/sh/megan_profile.lca.tsv.gz&quot;</span>)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   standardId = col_integer(),
##   MeganPercentual = col_double(),
##   MeganSequencesNumber = col_integer(),
##   MeganScoreMean = col_double(),
##   MeganScoreStandardDeviation = col_double(),
##   MeganScoreIQR = col_double(),
##   MeganScoreMedian = col_double(),
##   MeganScoreCI = col_double(),
##   rank = col_character()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meganProfileDf &lt;-<span class="st"> </span>meganProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(standardId <span class="op">%&gt;%</span><span class="st"> </span>is.na <span class="op">==</span><span class="st"> </span>F)
meganProfileDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>meganProfileDf<span class="op">$</span>standardId
meganProfileDf<span class="op">$</span>meganVote &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="kw">head</span>(meganProfileDf)</code></pre></div>
<pre><code>## # A tibble: 6 × 11
##   standardId MeganPercentual MeganSequencesNumber MeganScoreMean
##        &lt;int&gt;           &lt;dbl&gt;                &lt;int&gt;          &lt;dbl&gt;
## 1          2    9.986511e+01              1378522       99.84113
## 2       2157    7.244361e-05                    1       63.00000
## 3       2759    5.541936e-02                  765       87.63922
## 4      10239    4.563947e-03                   63       84.63492
## 5        976    1.825130e+01               251938       99.38970
## 6       1090    1.448872e-04                    2       75.00000
## # ... with 7 more variables: MeganScoreStandardDeviation &lt;dbl&gt;,
## #   MeganScoreIQR &lt;dbl&gt;, MeganScoreMedian &lt;dbl&gt;, MeganScoreCI &lt;dbl&gt;,
## #   rank &lt;chr&gt;, taxonomyId &lt;int&gt;, meganVote &lt;dbl&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(meganProfileDf)</code></pre></div>
<pre><code>## [1] 2656</code></pre>
</div>
<div id="gottcha" class="section level2">
<h2>GOTTCHA</h2>
<p>Functions</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loadRawGottchaProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">gottchaProfileLocation =</span> <span class="ot">NULL</span>) {
  rawProfileDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(gottchaProfileLocation)
  <span class="kw">return</span>(rawProfileDf)
}

loadGottchaProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">gottchaProfileLocation =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  gottchaProfile &lt;-<span class="st"> </span><span class="kw">loadRawGottchaProfile</span>(gottchaProfileLocation)
  <span class="kw">colnames</span>(gottchaProfile) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;rank&quot;</span>, <span class="st">&quot;taxonomyName&quot;</span>, <span class="st">&quot;GottchaPercentual&quot;</span>, <span class="st">&quot;GottchaLinearLength&quot;</span>, <span class="st">&quot;GottchaTotalBpMapped&quot;</span>, <span class="st">&quot;GottchaHitCount&quot;</span>, <span class="st">&quot;GottchaHitCountPlasmid&quot;</span>,
                                <span class="st">&quot;GottchaReadCount&quot;</span>, <span class="st">&quot;GottchaLinearDoc&quot;</span>, <span class="st">&quot;GottchaNormCov&quot;</span>)

  gottchaProfile<span class="op">$</span>standardId &lt;-<span class="st"> </span>gottchaProfile<span class="op">$</span>taxonomyName <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardTaxonomyIdsByNames</span>(taxdumprObject, .)
  gottchaProfile<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">getTaxonomyRanksByIds</span>(taxdumprObject, gottchaProfile<span class="op">$</span>standardId)
  gottchaProfile &lt;-<span class="st"> </span>gottchaProfile <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(rank <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;phylum&quot;</span>, <span class="st">&quot;class&quot;</span>, <span class="st">&quot;order&quot;</span>, <span class="st">&quot;family&quot;</span>, <span class="st">&quot;genus&quot;</span>, <span class="st">&quot;species&quot;</span>))

  <span class="co"># taxonomyDf &lt;- gottchaProfile$taxonomyId %&gt;% unique %&gt;% getStandardLineageIdsAndScientificNamesByIdsAsDataFrame(taxdumprObject, .)</span>
  <span class="co"># gottchaProfile &lt;- gottchaProfile %&gt;% merge(taxonomyDf, by = &quot;taxonomyId&quot;)</span>

  <span class="co"># speciesGottchaProfileForSh &lt;- gottchaProfile %&gt;%</span>
  <span class="co">#   subset(speciesId %&gt;% is.na == F) %&gt;%</span>
  <span class="co">#   mutate(standardId = speciesId, rank = &quot;species&quot;) %&gt;%</span>
  <span class="co">#   group_by(standardId, rank) %&gt;%</span>
  <span class="co">#   summarise(GottchaPercentual = sum(REL_ABUNDANCE))</span>
  <span class="co">#</span>
  <span class="co"># genusGottchaProfileForSh &lt;- gottchaProfile %&gt;%</span>
  <span class="co">#   subset(genusId %&gt;% is.na == F) %&gt;%</span>
  <span class="co">#   subset(speciesId %&gt;% is.na == T) %&gt;%</span>
  <span class="co">#   mutate(standardId = genusId, rank = &quot;genus&quot;) %&gt;%</span>
  <span class="co">#   group_by(standardId, rank) %&gt;%</span>
  <span class="co">#   summarise(GottchaPercentual = sum(REL_ABUNDANCE))</span>
  <span class="co">#</span>
  <span class="co"># gottchaProfile &lt;- rbind(genusGottchaProfileForSh, speciesGottchaProfileForSh)</span>

  <span class="kw">return</span>(gottchaProfile)
}

<span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh&quot;</span>, <span class="st">&quot;gottcha_profile.gottcha.tsv.gz&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;gottcha_profile.gottcha.tsv.gz&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gottchaProfileLocation &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh&quot;</span>, <span class="st">&quot;gottcha_profile.gottcha.tsv.gz&quot;</span>, <span class="dt">full.names =</span> T)
gottchaProfileDf &lt;-<span class="st"> </span><span class="kw">loadGottchaProfile</span>(gottchaProfileLocation, taxdumprObject)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   LEVEL = col_character(),
##   TAXA = col_character(),
##   REL_ABUNDANCE = col_double(),
##   LINEAR_LENGTH = col_integer(),
##   TOTAL_BP_MAPPED = col_integer(),
##   HIT_COUNT = col_integer(),
##   HIT_COUNT_PLASMID = col_integer(),
##   READ_COUNT = col_integer(),
##   LINEAR_DOC = col_double(),
##   NORM_COV = col_double()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gottchaProfileDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>gottchaProfileDf<span class="op">$</span>standardId
gottchaProfileDf<span class="op">$</span>gottchaVote &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="kw">head</span>(gottchaProfileDf)</code></pre></div>
<pre><code>## # A tibble: 6 × 13
##     rank   taxonomyName GottchaPercentual GottchaLinearLength
##    &lt;chr&gt;          &lt;chr&gt;             &lt;dbl&gt;               &lt;int&gt;
## 1 phylum     Firmicutes            0.3105             6105557
## 2  class Actinobacteria            0.3015             2385490
## 3 phylum Proteobacteria            0.2090             2692426
## 4 phylum  Bacteroidetes            0.1791             9958132
## 5  class        Bacilli            0.3105             6105557
## 6  class Actinobacteria            0.3015             2385490
## # ... with 9 more variables: GottchaTotalBpMapped &lt;int&gt;,
## #   GottchaHitCount &lt;int&gt;, GottchaHitCountPlasmid &lt;int&gt;,
## #   GottchaReadCount &lt;int&gt;, GottchaLinearDoc &lt;dbl&gt;, GottchaNormCov &lt;dbl&gt;,
## #   standardId &lt;dbl&gt;, taxonomyId &lt;dbl&gt;, gottchaVote &lt;dbl&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(gottchaProfileDf)</code></pre></div>
<pre><code>## [1] 40</code></pre>
</div>
<div id="kraken" class="section level2">
<h2>Kraken</h2>
<p>Functions to load Kraken Report</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># loadKrakenProfile &lt;- function(krakenReportLocation = NULL, krakenScoresLocation = NULL, taxdumprObject = NULL) {</span>
<span class="co">#   KRAKEN_REPORT_COLUMNS_NAMES &lt;- c(&quot;percentageOfReads&quot;, &quot;cladeNumberOfReads&quot;, &quot;taxonNumberOfReads&quot;, &quot;rank&quot;, &quot;taxonomyId&quot;, &quot;taxonomyName&quot;)</span>
<span class="co">#   # profile &lt;- read.table(krakenReportLocation, header = F, sep = &quot;\t&quot;, &quot;&quot;)</span>
<span class="co">#   profile &lt;- readr::read_tsv(krakenReportLocation, col_names = F)</span>
<span class="co">#   colnames(profile) &lt;- KRAKEN_REPORT_COLUMNS_NAMES</span>
<span class="co">#</span>
<span class="co">#   profile$taxonomyId &lt;- getUpdatedIds(taxdumprObject, profile$taxonomyId)</span>
<span class="co">#   # profile$standardId &lt;- getStandardTaxonomyIdsByIds(taxdumprObject, profile$taxonomyId)</span>
<span class="co">#   # profile$standardId[profile$standardId %&gt;% as.numeric %&gt;% is.na] &lt;-</span>
<span class="co">#   #   getStandardTaxonomyIdsByNames(taxdumprObject, profile$taxonomyName[profile$standardId %&gt;% as.numeric %&gt;% is.na])</span>
<span class="co">#   # profile$taxonomyName &lt;- getScientificNamesByIds(taxdumprObject, profile$taxonomyId)</span>
<span class="co">#   profile$rank &lt;- getTaxonomyRanksByIds(taxdumprObject, profile$taxonomyId)</span>
<span class="co">#</span>
<span class="co">#   profile &lt;- profile %&gt;%</span>
<span class="co">#     subset(rank %in% c(&quot;phylum&quot;, &quot;class&quot;, &quot;order&quot;, &quot;family&quot;, &quot;genus&quot;, &quot;species&quot;)) %&gt;%</span>
<span class="co">#     mutate(standardId = taxonomyId) %&gt;%</span>
<span class="co">#     group_by(rank, standardId) %&gt;%</span>
<span class="co">#     summarize(percentageOfReads = sum(percentageOfReads),</span>
<span class="co">#               cladeNumberOfReads = sum(cladeNumberOfReads),</span>
<span class="co">#               taxonNumberOfReads = sum(taxonNumberOfReads))</span>
<span class="co">#</span>
<span class="co">#   numberOfQualitySequences &lt;- nrow(readr::read_tsv(krakenScoresLocation, col_names = F))</span>
<span class="co">#   profile$KrakenPercentual &lt;- profile$cladeNumberOfReads / numberOfQualitySequences * 100</span>
<span class="co">#</span>
<span class="co">#   return(profile)</span>
<span class="co"># }</span>

loadRawKrakenProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">krakenReportLocation =</span> <span class="ot">NULL</span>) {
  KRAKEN_REPORT_COLUMNS_NAMES &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;percentageOfReads&quot;</span>, <span class="st">&quot;cladeNumberOfReads&quot;</span>, <span class="st">&quot;taxonNumberOfReads&quot;</span>, <span class="st">&quot;rank&quot;</span>, <span class="st">&quot;taxonomyId&quot;</span>, <span class="st">&quot;taxonomyName&quot;</span>)
  <span class="co"># profile &lt;- read.table(krakenReportLocation, header = F, sep = &quot;\t&quot;, &quot;&quot;)</span>
  profile &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(krakenReportLocation, <span class="dt">col_names =</span> F)
  <span class="kw">colnames</span>(profile) &lt;-<span class="st"> </span>KRAKEN_REPORT_COLUMNS_NAMES
  <span class="kw">return</span>(profile)
}

processRawKrakenProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">rawKrakenProfile =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  RANKS &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;K&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;O&quot;</span>, <span class="st">&quot;F&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;S&quot;</span>)
  krakenProfile &lt;-<span class="st"> </span>rawKrakenProfile <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">subset</span>(rank <span class="op">%in%</span><span class="st"> </span>RANKS)

  krakenProfile<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>krakenProfile<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">getUpdatedIds</span>(taxdumprObject, .)

  ## <span class="al">TODO</span>: check duplications

  <span class="kw">return</span>(krakenProfile)
}

<span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh&quot;</span>, <span class="st">&quot;kraken_report.filtered.tsv.gz&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;kraken_report.filtered.tsv.gz&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">krakenReportLocation &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh&quot;</span>, <span class="st">&quot;kraken_report.filtered.tsv.gz&quot;</span>, <span class="dt">full.names =</span> T)
rawKrakenProfileDf &lt;-<span class="st"> </span><span class="kw">loadRawKrakenProfile</span>(<span class="dt">krakenReportLocation =</span> krakenReportLocation)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   X1 = col_double(),
##   X2 = col_integer(),
##   X3 = col_integer(),
##   X4 = col_character(),
##   X5 = col_integer(),
##   X6 = col_character()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(rawKrakenProfileDf)</code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   percentageOfReads cladeNumberOfReads taxonNumberOfReads  rank taxonomyId
##               &lt;dbl&gt;              &lt;int&gt;              &lt;int&gt; &lt;chr&gt;      &lt;int&gt;
## 1              7.33             114396             114396     U          0
## 2             92.67            1446765              12926     -          1
## 3             91.84            1433777                  3     -     131567
## 4             91.84            1433774               2103     D          2
## 5             32.70             510547               1101     P       1239
## 6             32.37             505342                244     C      91061
## # ... with 1 more variables: taxonomyName &lt;chr&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(rawKrakenProfileDf)</code></pre></div>
<pre><code>## [1] 528</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">krakenProfileDf &lt;-<span class="st"> </span><span class="kw">processRawKrakenProfile</span>(rawKrakenProfileDf, taxdumprObject)
krakenProfileDf<span class="op">$</span>krakenVote &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="kw">head</span>(krakenProfileDf)</code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   percentageOfReads cladeNumberOfReads taxonNumberOfReads  rank taxonomyId
##               &lt;dbl&gt;              &lt;int&gt;              &lt;int&gt; &lt;chr&gt;      &lt;int&gt;
## 1             32.70             510547               1101     P       1239
## 2             32.37             505342                244     C      91061
## 3             32.35             505073               1578     O     186826
## 4             24.55             383203               1535     F      33958
## 5             24.41             381139              16875     G       1578
## 6              9.76             152438             131774     S       1590
## # ... with 2 more variables: taxonomyName &lt;chr&gt;, krakenVote &lt;dbl&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(krakenProfileDf)</code></pre></div>
<pre><code>## [1] 236</code></pre>
</div>
<div id="lmat" class="section level2">
<h2>LMAT</h2>
<p>Functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loadRawLmatProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">lmatProfileLocation =</span> <span class="ot">NULL</span>) {
  lmatProfile &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(lmatProfileLocation, <span class="dt">col_names =</span> F)
  <span class="kw">colnames</span>(lmatProfile) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;scoresSum&quot;</span>, <span class="st">&quot;readsNumber&quot;</span>, <span class="st">&quot;taxonomyId&quot;</span>, <span class="st">&quot;taxonomyDescription&quot;</span>)
  <span class="kw">return</span>(lmatProfile)
}

processLmatProfileTaxonomy &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">rawLmatProfile =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  taxonomyDf &lt;-<span class="st"> </span><span class="kw">unique</span>(rawLmatProfile<span class="op">$</span>taxonomyId) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardLineageIdsAndScientificNamesByIdsAsDataFrame</span>(taxdumprObject, .)
  lmatProfile &lt;-<span class="st"> </span>rawLmatProfile <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">merge</span>(taxonomyDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all.x =</span> T)
  lmatProfile<span class="op">$</span>rank &lt;-<span class="st"> </span>lmatProfile<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getTaxonomyRanksByIds</span>(taxdumprObject, .)
  <span class="kw">return</span>(lmatProfile)
}

<span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh/lmat_profile/&quot;</span>, <span class="st">&quot;qc_filtered.fastq.lmat-4-14.20mer.db.lo.rl_output.0.30.fastsummary$&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;qc_filtered.fastq.lmat-4-14.20mer.db.lo.rl_output.0.30.fastsummary&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lmatProfileLocation &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh/lmat_profile/&quot;</span>, <span class="st">&quot;qc_filtered.fastq.lmat-4-14.20mer.db.lo.rl_output.0.30.fastsummary$&quot;</span>, <span class="dt">full.names =</span> T)
rawLmatProfileDf &lt;-<span class="st"> </span><span class="kw">loadRawLmatProfile</span>(lmatProfileLocation)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   X1 = col_double(),
##   X2 = col_integer(),
##   X3 = col_integer(),
##   X4 = col_character()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lmatProfileDf &lt;-<span class="st"> </span><span class="kw">processLmatProfileTaxonomy</span>(rawLmatProfileDf, taxdumprObject)
<span class="kw">head</span>(lmatProfileDf)</code></pre></div>
<pre><code>##   taxonomyId   scoresSum readsNumber                 taxonomyDescription
## 1          1 18481.30000       18950 depth=0,taxid=1,ktaxid=1,entries=-1
## 2          2 49991.60000       60479               superkingdom,Bacteria
## 3        270    11.73710          14                       genus,Thermus
## 4        274     9.29873          11        species,Thermus thermophilus
## 5        286    33.56810          51                   genus,Pseudomonas
## 6        294    17.62760          31     species,Pseudomonas fluorescens
##   superkingdomId phylumId classId orderId familyId genusId speciesId
## 1             NA       NA      NA      NA       NA      NA        NA
## 2              2       NA      NA      NA       NA      NA        NA
## 3              2     1297  188787   68933   188786     270        NA
## 4              2     1297  188787   68933   188786     270       274
## 5              2     1224    1236   72274   135621     286        NA
## 6              2     1224    1236   72274   135621     286       294
##              taxonomyName superkingdomName          phylumName
## 1                    root             &lt;NA&gt;                &lt;NA&gt;
## 2                Bacteria         Bacteria                &lt;NA&gt;
## 3                 Thermus         Bacteria Deinococcus-Thermus
## 4    Thermus thermophilus         Bacteria Deinococcus-Thermus
## 5             Pseudomonas         Bacteria      Proteobacteria
## 6 Pseudomonas fluorescens         Bacteria      Proteobacteria
##             className       orderName       familyName   genusName
## 1                &lt;NA&gt;            &lt;NA&gt;             &lt;NA&gt;        &lt;NA&gt;
## 2                &lt;NA&gt;            &lt;NA&gt;             &lt;NA&gt;        &lt;NA&gt;
## 3          Deinococci       Thermales       Thermaceae     Thermus
## 4          Deinococci       Thermales       Thermaceae     Thermus
## 5 Gammaproteobacteria Pseudomonadales Pseudomonadaceae Pseudomonas
## 6 Gammaproteobacteria Pseudomonadales Pseudomonadaceae Pseudomonas
##               speciesName         rank
## 1                    &lt;NA&gt;         &lt;NA&gt;
## 2                    &lt;NA&gt; superkingdom
## 3                    &lt;NA&gt;        genus
## 4    Thermus thermophilus      species
## 5                    &lt;NA&gt;        genus
## 6 Pseudomonas fluorescens      species</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(lmatProfileDf)</code></pre></div>
<pre><code>## [1] 742</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lmatProfileTaxonomyIds &lt;-<span class="st"> </span>lmatProfileDf[,<span class="dv">5</span><span class="op">:</span><span class="dv">11</span>] <span class="op">%&gt;%</span><span class="st"> </span>unlist <span class="op">%&gt;%</span><span class="st"> </span>unique
lmatProfileDf &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">taxonomyId =</span> lmatProfileTaxonomyIds)
lmatProfileDf<span class="op">$</span>lmatVote &lt;-<span class="st"> </span><span class="dv">1</span>
lmatProfileDf &lt;-<span class="st"> </span>lmatProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(taxonomyId <span class="op">%&gt;%</span><span class="st"> </span>is.na <span class="op">==</span><span class="st"> </span>F)
<span class="kw">head</span>(lmatProfileDf)</code></pre></div>
<pre><code>##   taxonomyId lmatVote
## 2          2        1
## 3       2157        1
## 4       2759        1
## 5      10239        1
## 6       1297        1
## 7       1224        1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(lmatProfileDf)</code></pre></div>
<pre><code>## [1] 787</code></pre>
</div>
<div id="metaphlan2" class="section level2">
<h2>Metaphlan2</h2>
<p>Functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loadMetaphlan2Profile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">profileLocation =</span> <span class="ot">NULL</span>, <span class="dt">taxdumprObject =</span> <span class="ot">NULL</span>) {
  profileDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(profileLocation)
  <span class="kw">colnames</span>(profileDf) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;lineage&quot;</span>, <span class="st">&quot;Metaphlan2Percentual&quot;</span>)

  ## Remove Viruses and subspecies classifications
  profileDf &lt;-<span class="st"> </span>profileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(<span class="kw">grepl</span>(<span class="st">&quot;k__Viruses&quot;</span>, lineage) <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>)
  profileDf &lt;-<span class="st"> </span>profileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(<span class="kw">grepl</span>(<span class="st">&quot;t__&quot;</span>, lineage) <span class="op">==</span><span class="st"> </span><span class="ot">FALSE</span>)

  ## Keep leafs only
  profileDf<span class="op">$</span>lineage &lt;-<span class="st"> </span>profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_remove</span>(<span class="st">&quot;.*</span><span class="ch">\\</span><span class="st">|&quot;</span>)

  ## Add ranks
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> &quot;&quot;</span>
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;k__&quot;</span>), <span class="st">&quot;superkingdom&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;p__&quot;</span>), <span class="st">&quot;phylum&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;c__&quot;</span>), <span class="st">&quot;class&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;o__&quot;</span>), <span class="st">&quot;order&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;f__&quot;</span>), <span class="st">&quot;family&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;g__&quot;</span>), <span class="st">&quot;genus&quot;</span>, profileDf<span class="op">$</span>rank)
  profileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="kw">ifelse</span>(profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_detect</span>(<span class="st">&quot;s__&quot;</span>), <span class="st">&quot;species&quot;</span>, profileDf<span class="op">$</span>rank)

  ## Get taxonomy names
  profileDf<span class="op">$</span>taxonomyName &lt;-<span class="st"> </span>profileDf<span class="op">$</span>lineage <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_remove</span>(<span class="st">&quot;[a-z]__&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace_all</span>(<span class="st">&quot;_&quot;</span>, <span class="st">&quot; &quot;</span>)

  ## Taxonomy id
  profileDf<span class="op">$</span>standardId &lt;-<span class="st"> </span>profileDf<span class="op">$</span>taxonomyName <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardTaxonomyIdsByNames</span>(taxdumprObject, .)
  profileDf<span class="op">$</span>standardId[profileDf<span class="op">$</span>standardId <span class="op">%&gt;%</span><span class="st"> </span>is.na] &lt;-<span class="st"> </span>profileDf<span class="op">$</span>taxonomyName[profileDf<span class="op">$</span>standardId <span class="op">%&gt;%</span><span class="st"> </span>is.na]

  ## Summarize
  profileDf &lt;-<span class="st"> </span>profileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(rank, standardId) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">Metaphlan2Percentual =</span> <span class="kw">sum</span>(Metaphlan2Percentual))

  <span class="kw">return</span>(profileDf)
}

<span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh/&quot;</span>, <span class="st">&quot;metaphlan2_profile.tsv.gz&quot;</span>)</code></pre></div>
<pre><code>## [1] &quot;metaphlan2_profile.tsv.gz&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">metaphlan2ProfileLocation &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data-raw/sh/&quot;</span>, <span class="st">&quot;metaphlan2_profile.tsv.gz&quot;</span>, <span class="dt">full.names =</span> T)
metaphlan2ProfileDf &lt;-<span class="st"> </span><span class="kw">loadMetaphlan2Profile</span>(metaphlan2ProfileLocation, taxdumprObject)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   `#SampleID` = col_character(),
##   Metaphlan2_Analysis = col_double()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(metaphlan2ProfileDf)</code></pre></div>
<pre><code>## Source: local data frame [6 x 3]
## Groups: rank [2]
## 
##     rank standardId Metaphlan2Percentual
##    &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;
## 1  class       1236             15.02328
## 2  class       1760             26.60764
## 3  class     186801              5.64674
## 4  class     200643             10.73170
## 5  class      91061             41.99065
## 6 family       1300             12.60466</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">metaphlan2ProfileDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>metaphlan2ProfileDf<span class="op">$</span>standardId
metaphlan2ProfileDf<span class="op">$</span>metaphlan2Vote &lt;-<span class="st"> </span><span class="dv">1</span>
<span class="kw">head</span>(metaphlan2ProfileDf)</code></pre></div>
<pre><code>## Source: local data frame [6 x 5]
## Groups: rank [2]
## 
##     rank standardId Metaphlan2Percentual taxonomyId metaphlan2Vote
##    &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;      &lt;chr&gt;          &lt;dbl&gt;
## 1  class       1236             15.02328       1236              1
## 2  class       1760             26.60764       1760              1
## 3  class     186801              5.64674     186801              1
## 4  class     200643             10.73170     200643              1
## 5  class      91061             41.99065      91061              1
## 6 family       1300             12.60466       1300              1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(metaphlan2ProfileDf)</code></pre></div>
<pre><code>## [1] 42</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(metaphlan2ProfileDf<span class="op">$</span>Metaphlan2Percentual)</code></pre></div>
<pre><code>## [1] 700</code></pre>
</div>
</div>
<div id="majority-vote" class="section level1">
<h1>Majority Vote</h1>
<div id="merging" class="section level2">
<h2>Merging</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subMeganProfile &lt;-<span class="st"> </span>meganProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(taxonomyId, meganVote)
subGottchaProfileDf &lt;-<span class="st"> </span>gottchaProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(taxonomyId, gottchaVote)
subKrakenProfileDf &lt;-<span class="st"> </span>krakenProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(taxonomyId, krakenVote)
subLmatProfileDf &lt;-<span class="st"> </span>lmatProfileDf
subMetaphlan2ProfileDf &lt;-<span class="st"> </span>metaphlan2ProfileDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(taxonomyId, metaphlan2Vote)</code></pre></div>
<pre><code>## Adding missing grouping variables: `rank`</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subMetaphlan2ProfileDf<span class="op">$</span>rank &lt;-<span class="st"> </span><span class="ot">NULL</span>

mergedDf &lt;-<span class="st"> </span><span class="kw">merge</span>(subMeganProfile, subGottchaProfileDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all =</span> T)
mergedDf &lt;-<span class="st"> </span><span class="kw">merge</span>(mergedDf, subKrakenProfileDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all =</span> T)
mergedDf &lt;-<span class="st"> </span><span class="kw">merge</span>(mergedDf, subLmatProfileDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all =</span> T)
mergedDf &lt;-<span class="st"> </span><span class="kw">merge</span>(mergedDf, subMetaphlan2ProfileDf, <span class="dt">by =</span> <span class="st">&quot;taxonomyId&quot;</span>, <span class="dt">all =</span> T)

mergedDf[mergedDf <span class="op">%&gt;%</span><span class="st"> </span>is.na] &lt;-<span class="st"> </span><span class="dv">0</span></code></pre></div>
</div>
<div id="voting" class="section level2">
<h2>Voting</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># majorityVoteDf &lt;- mergedDf %&gt;% mutate(votesSum = meganVote + gottchaVote + krakenVote + lmatVote + metaphlan2Vote) %&gt;% subset(votesSum &gt;= 3)</span>
majorityVoteDf &lt;-<span class="st"> </span>mergedDf <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">votesSum =</span> meganVote <span class="op">+</span><span class="st"> </span>gottchaVote <span class="op">+</span><span class="st"> </span>krakenVote <span class="op">+</span><span class="st"> </span>lmatVote <span class="op">+</span><span class="st"> </span>metaphlan2Vote) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">votesClassification =</span> votesSum <span class="op">&gt;=</span><span class="st"> </span><span class="dv">3</span>)
majorityVoteDf<span class="op">$</span>rank &lt;-<span class="st"> </span>majorityVoteDf<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span>as.numeric <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getTaxonomyRanksByIds</span>(taxdumprObject, .)</code></pre></div>
<pre><code>## Warning in function_list[[i]](value): NAs introduzidos por coerção</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(majorityVoteDf)</code></pre></div>
<pre><code>##   taxonomyId meganVote gottchaVote krakenVote lmatVote metaphlan2Vote
## 1     100174         1           0          0        0              0
## 2      10028         1           0          0        0              0
## 3      10029         1           0          0        0              0
## 4      10035         1           0          0        0              0
## 5      10036         1           0          0        0              0
## 6     100468         1           0          0        0              0
##   votesSum votesClassification    rank
## 1        1               FALSE species
## 2        1               FALSE   genus
## 3        1               FALSE species
## 4        1               FALSE   genus
## 5        1               FALSE species
## 6        1               FALSE species</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">write.table</span>(majorityVoteDf, <span class="kw">gzfile</span>(<span class="st">&quot;majority_vote.tsv.gz&quot;</span>), <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote =</span> F, <span class="dt">row.names =</span> F, <span class="dt">col.names =</span> T)</code></pre></div>
</div>
<div id="evaluation" class="section level2">
<h2>Evaluation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">referenceMetadaDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(<span class="st">&quot;../data-raw/referenceMetadata.tsv&quot;</span>)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   searchTerm = col_character(),
##   relativeAbundance = col_double(),
##   taxid = col_integer(),
##   organism = col_character(),
##   speciestaxid = col_integer(),
##   speciesname = col_character(),
##   ftppath_genbank = col_character(),
##   ftppath_refseq = col_character(),
##   usedFtpPathForGenome = col_character(),
##   usedFtpPathForCdss = col_character(),
##   genomeCheck = col_logical(),
##   cdsCheck = col_logical(),
##   downloadStatus = col_logical(),
##   prodigalCds = col_logical(),
##   primerCompatibleGenome = col_logical(),
##   relativeAbundanceForAmplicons = col_integer()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">referenceMetadaDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>referenceMetadaDf<span class="op">$</span>taxid
referenceMetadaDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>referenceMetadaDf<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getUpdatedIds</span>(taxdumprObject, .)
referenceTaxonomyDf &lt;-<span class="st"> </span>referenceMetadaDf<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">getStandardLineageIdsAndScientificNamesByIdsAsDataFrame</span>(taxdumprObject, .)</code></pre></div>
<ul>
<li>Species Level:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">majorityVoteSpeciesDf &lt;-<span class="st"> </span>majorityVoteDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(rank <span class="op">==</span><span class="st"> &quot;species&quot;</span>)
<span class="kw">table</span>(majorityVoteSpeciesDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>speciesId)</code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  1989    11</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mean</span>(majorityVoteSpeciesDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>speciesId)</code></pre></div>
<pre><code>## [1] 0.0055</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">speciesTps &lt;-<span class="st"> </span><span class="kw">sum</span>(majorityVoteSpeciesDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>speciesId)
speciesFps &lt;-<span class="st"> </span><span class="kw">sum</span>(majorityVoteSpeciesDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>speciesId <span class="op">==</span><span class="st"> </span>F)
speciesFns &lt;-<span class="st"> </span><span class="kw">sum</span>(referenceTaxonomyDf<span class="op">$</span>speciesId <span class="op">%in%</span><span class="st"> </span>majorityVoteSpeciesDf<span class="op">$</span>taxonomyId <span class="op">==</span><span class="st"> </span>F)
speciesPrecision &lt;-<span class="st"> </span>speciesTps <span class="op">/</span><span class="st"> </span>(speciesTps <span class="op">+</span><span class="st"> </span>speciesFps)
speciesRecall &lt;-<span class="st"> </span>speciesTps <span class="op">/</span><span class="st"> </span>(speciesTps <span class="op">+</span><span class="st"> </span>speciesFns)
speciesF1 &lt;-<span class="st"> </span><span class="dv">2</span> <span class="op">*</span><span class="st"> </span>speciesPrecision <span class="op">*</span><span class="st"> </span>speciesRecall <span class="op">/</span><span class="st"> </span>(speciesPrecision <span class="op">+</span><span class="st"> </span>speciesRecall)
speciesPrecision</code></pre></div>
<pre><code>## [1] 0.0055</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">speciesRecall</code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">speciesF1</code></pre></div>
<pre><code>## [1] 0.01093983</code></pre>
<ul>
<li>Genus Level:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">majorityVoteGenusDf &lt;-<span class="st"> </span>majorityVoteDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(rank <span class="op">==</span><span class="st"> &quot;genus&quot;</span>)
<span class="kw">table</span>(majorityVoteGenusDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>genusId)</code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##    18     6</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mean</span>(majorityVoteGenusDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>genusId)</code></pre></div>
<pre><code>## [1] 0.25</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genusTps &lt;-<span class="st"> </span><span class="kw">sum</span>(majorityVoteGenusDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>genusId)
genusFps &lt;-<span class="st"> </span><span class="kw">sum</span>(majorityVoteGenusDf<span class="op">$</span>taxonomyId <span class="op">%in%</span><span class="st"> </span>referenceTaxonomyDf<span class="op">$</span>genusId <span class="op">==</span><span class="st"> </span>F)
genusFns &lt;-<span class="st"> </span><span class="kw">sum</span>(referenceTaxonomyDf<span class="op">$</span>genusId <span class="op">%in%</span><span class="st"> </span>majorityVoteGenusDf<span class="op">$</span>taxonomyId <span class="op">==</span><span class="st"> </span>F)
genusPrecision &lt;-<span class="st"> </span>genusTps <span class="op">/</span><span class="st"> </span>(genusTps <span class="op">+</span><span class="st"> </span>genusFps)
genusRecall &lt;-<span class="st"> </span>genusTps <span class="op">/</span><span class="st"> </span>(genusTps <span class="op">+</span><span class="st"> </span>genusFns)
genusF1 &lt;-<span class="st"> </span>(<span class="dv">2</span> <span class="op">*</span><span class="st"> </span>genusPrecision <span class="op">*</span><span class="st"> </span>genusRecall) <span class="op">/</span><span class="st"> </span>(genusPrecision <span class="op">+</span><span class="st"> </span>genusRecall)
genusPrecision</code></pre></div>
<pre><code>## [1] 0.25</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genusRecall</code></pre></div>
<pre><code>## [1] 0.8571429</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genusF1</code></pre></div>
<pre><code>## [1] 0.3870968</code></pre>
</div>
</div>
<div id="functions-to-load-majority-vote-profile" class="section level1">
<h1>Functions to load majority vote profile</h1>
<div id="load-raw" class="section level2">
<h2>Load RAW</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">loadRawMajorityVoteProfile &lt;-<span class="st"> </span><span class="cf">function</span>(<span class="dt">profileLocation =</span> <span class="ot">NULL</span>) {
  COLUMNS_TYPES &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">cols</span>(
    readr<span class="op">::</span><span class="kw">col_character</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_integer</span>(),
    readr<span class="op">::</span><span class="kw">col_logical</span>(),
    readr<span class="op">::</span><span class="kw">col_character</span>())
  profileDf &lt;-<span class="st"> </span>readr<span class="op">::</span><span class="kw">read_tsv</span>(profileLocation, <span class="dt">col_types =</span> COLUMNS_TYPES)
}
testDf &lt;-<span class="st"> </span><span class="kw">loadRawMajorityVoteProfile</span>(<span class="st">&quot;../data-raw/sh/majority_vote.tsv.gz&quot;</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(testDf))</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">taxonomyId</th>
<th align="right">meganVote</th>
<th align="right">gottchaVote</th>
<th align="right">krakenVote</th>
<th align="right">lmatVote</th>
<th align="right">metaphlan2Vote</th>
<th align="right">votesSum</th>
<th align="left">votesClassification</th>
<th align="left">rank</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">100174</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">species</td>
</tr>
<tr class="even">
<td align="left">10028</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">genus</td>
</tr>
<tr class="odd">
<td align="left">10029</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">species</td>
</tr>
<tr class="even">
<td align="left">10035</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">genus</td>
</tr>
<tr class="odd">
<td align="left">10036</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">species</td>
</tr>
<tr class="even">
<td align="left">100468</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">FALSE</td>
<td align="left">species</td>
</tr>
</tbody>
</table>
</div>
<div id="performance-evaluation" class="section level2">
<h2>Performance Evaluation</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">testDf &lt;-<span class="st"> </span>testDf <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">subset</span>(taxonomyId <span class="op">%&gt;%</span><span class="st"> </span>as.numeric <span class="op">%&gt;%</span><span class="st"> </span>is.na <span class="op">==</span><span class="st"> </span>F)</code></pre></div>
<pre><code>## Warning in function_list[[i]](value): NAs introduzidos por coerção</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">testDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span>testDf<span class="op">$</span>taxonomyId <span class="op">%&gt;%</span><span class="st"> </span>as.numeric
testDf<span class="op">$</span>taxonomyId &lt;-<span class="st"> </span><span class="kw">getUpdatedIds</span>(taxdumprObject, testDf<span class="op">$</span>taxonomyId)
testDf<span class="op">$</span>standardId &lt;-<span class="st"> </span>testDf<span class="op">$</span>taxonomyId
<span class="kw">head</span>(testDf)</code></pre></div>
<pre><code>## # A tibble: 6 × 10
##   taxonomyId meganVote gottchaVote krakenVote lmatVote metaphlan2Vote
##        &lt;dbl&gt;     &lt;int&gt;       &lt;int&gt;      &lt;int&gt;    &lt;int&gt;          &lt;int&gt;
## 1     100174         1           0          0        0              0
## 2      10028         1           0          0        0              0
## 3      10029         1           0          0        0              0
## 4      10035         1           0          0        0              0
## 5      10036         1           0          0        0              0
## 6     100468         1           0          0        0              0
## # ... with 4 more variables: votesSum &lt;int&gt;, votesClassification &lt;lgl&gt;,
## #   rank &lt;chr&gt;, standardId &lt;dbl&gt;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">majorityVoteSpeciesDf &lt;-<span class="st"> </span>testDf <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">subset</span>(rank <span class="op">==</span><span class="st"> &quot;species&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">subset</span>(votesClassification) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">select</span>(<span class="dt">speciesId =</span> standardId, <span class="dt">predicted =</span> votesClassification)
referenceSpeciesDf &lt;-<span class="st"> </span>referenceTaxonomyDf <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">select</span>(speciesId) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">actual =</span> <span class="ot">TRUE</span>)
speciesEvaluationDf &lt;-<span class="st"> </span><span class="kw">merge</span>(referenceSpeciesDf, majorityVoteSpeciesDf, <span class="dt">all =</span> T)
speciesEvaluationDf[speciesEvaluationDf <span class="op">%&gt;%</span><span class="st"> </span>is.na] &lt;-<span class="st"> </span><span class="ot">FALSE</span>
Metrics<span class="op">::</span><span class="kw">auc</span>(speciesEvaluationDf<span class="op">$</span>actual, speciesEvaluationDf<span class="op">$</span>predicted)</code></pre></div>
<pre><code>## [1] 0.5</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Metrics<span class="op">::</span><span class="kw">precision</span>(speciesEvaluationDf<span class="op">$</span>actual, speciesEvaluationDf<span class="op">$</span>predicted)</code></pre></div>
<pre><code>## [1] 0.2181818</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Metrics<span class="op">::</span><span class="kw">recall</span>(speciesEvaluationDf<span class="op">$</span>actual, speciesEvaluationDf<span class="op">$</span>predicted)</code></pre></div>
<pre><code>## [1] 1</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Metrics<span class="op">::</span><span class="kw">accuracy</span>(speciesEvaluationDf<span class="op">$</span>actual, speciesEvaluationDf<span class="op">$</span>predicted)</code></pre></div>
<pre><code>## [1] 0.2181818</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Metrics<span class="op">::</span><span class="kw">f1</span>(speciesEvaluationDf<span class="op">$</span>actual, speciesEvaluationDf<span class="op">$</span>predicted)</code></pre></div>
<pre><code>## [1] 0.6666667</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">View</span>(speciesEvaluationDf)</code></pre></div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
